torchdynamo/guards.py

```diff
@@ -52,6 +60,85 @@
 )
 
 
+class NNModuleChangeTrackerUtil:
+    original_register_parameter = None
+    original_add_module = None
+    original_load_state_dict = None
+    original_setattr = None
+
+    @classmethod
+    def register_patches_on_torch_nn_module(cls):
+        if getattr(
+            torch.nn.Module, "_NNModuleChangeTrackerUtil__dynamo_module_patch", True
+        ):
+            # register_parameter
+            torch.nn.Module.__dynamo_module_patch = False
```

Comment:
Should this be `_NNModuleChangeTrackerUtil__dynamo_module_patch`?

Reply:
No - this becomes `_NNModuleChangeTrackerUtil__dynamo_module_patch`

Reply:
Actually, do you know why the name is getting mangled with the calling class like this?

Reply:
I have no idea what could be causing this name mangling, or if we can count on the name mangling happening in the future.

Reply:
Discussed offline, this is a "feature" of python.